Failed to build 5.2 release on ubuntu 20.04

I downloaded 5.2 from github and untar-ed it to a directory, run build.sh and I got this

```
inlined from ‘getVersion’ at /home/test/FTL-5.2/src/api/api.c:1062:2:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:34:10: error: ‘__builtin_memcpy’ forming offset [2, 7] is out of the bounds [0, 1] [-Werror=array-bounds]
   34 |   return __builtin___memcpy_chk (__dest, __src, __len, __bos0 (__dest));
```

I then git clone the git repository and it built fine.

The problem is with src/version.h.in, for released files those variables are empty, and gcc/memcpy does not like that.

maybe in the release version.in.h file it should be pre-filled release info manually? after I manually put stuff in I can build fine.

--- a/src/api/api.c
+++ b/src/api/api.c
@@ -1174,8 +1174,8 @@ void getVersion(const int *sock)
 	const char *version = get_FTL_version();
 
 	// Extract first 7 characters of the hash
-	char hash[8];
-	memcpy(hash, commit, 7); hash[7] = 0;
+	char hash[8] = { 0 };
+	memcpy(hash, commit, min((size_t)7, strlen(commit)));
 
 	if(strlen(tag) > 1) {
 		if(istelnet[*sock])

--- a/src/log.c
+++ b/src/log.c
@@ -286,15 +286,22 @@ const char __attribute__ ((malloc)) *get_FTL_version(void)
 	// Obtain FTL version if not already determined
 	if(FTLversion == NULL)
 	{
-		if(strlen(GIT_TAG) > 1)
+		if(strlen(GIT_TAG) > 1 && strlen(GIT_VERSION) > 1)
 		{
+			// Copy version string if this is a tagged release
 			FTLversion = strdup(GIT_VERSION);
 		}
-		else
+		else if(strlen(GIT_HASH) > 0)
 		{
+			// Build special version string when there is a hash
 			FTLversion = calloc(13, sizeof(char));
 			// Build version by appending 7 characters of the hash to "vDev-"
 			snprintf(FTLversion, 13, "vDev-%.7s", GIT_HASH);
+		}
+		else
+		{
+			// Fallback for tarball build, etc. without any GIT subsystem
+			FTLversion = strdup("UNKNOWN (not a GIT build)");
 		}
 	}
 

